packages/iov-bns/src/bnsconnection.ts

```diff
@@ -558,28 +558,20 @@ export class BnsConnection implements BcpAtomicSwapConnection {
   }
 
   public watchBlockHeaders(): Stream<BlockHeader> {
-    // TODO: this implementation is crazy but currently we have no way to calculate a
-    // block ID from a block header
-
-    let headersSubscription: Subscription;
-    // create explicit producer because Steam.map() does not work with async function
-    const producer: Producer<BlockHeader> = {
-      start: listener => {
-        headersSubscription = this.tmClient.subscribeNewBlockHeader().subscribe({
-          next: header => {
-            this.getBlockHeader(header.height)
-              .then(blockHeader => listener.next(blockHeader))
-              .catch(error => listener.error(error));
-          },
-          error: error => listener.error(error),
-          complete: () => listener.error("Source stream completed"),
-        });
-      },
-      stop: () => {
-        headersSubscription.unsubscribe();
-      },
-    };
-    return Stream.create(producer);
+    // TODO: ID unavailable because
+    // - we cannot trivially calculate it https://github.com/iov-one/iov-core/issues/618 and
+    // - want to avoid an extra query to the node which causes issues when trying to send to a disconnected socket
+    // Leave it empty as long as no application strictly requires the ID
+    const dummyBlockId = "" as BlockId;
```

Comment:
I would make it clear it is dummy.
If a caller tries it and it is empty, they will wonder
If they try and it say `Id not yet implemented` or such, it will be clear

Reply:
jap, good idea, done